### PR TITLE
Add Callable function to override default language

### PR DESCRIPTION
**Pull request by jerome-fix**
refs #182

### CODE REVIEW

File: src/ORM/Translatable/DefaultLocaleCallable.php [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+use Symfony\Component\DependencyInjection\Container;
+
+/**
+ * @author    Jérôme Fix <jerome.fix@zapoyok.info>
+ */
+class DefaultLocaleCallable
+{
+    private $container;
+
+    public function __construct(Container $container)
```
Comment (docteurklein) [inline src/ORM/Translatable/DefaultLocaleCallable.php]: you should inject the parameter directly.

File: config/orm-services.yml [hunk 2]
```diff
@@ -51,6 +53,12 @@ services:
             - "@service_container" # lazy request resolution
         public:  false
 
+    knp.doctrine_behaviors.translatable_subscriber.default_locale_callable:
+        class:           "%knp.doctrine_behaviors.translatable_subscriber.default_locale_callable.class%"
+        arguments:
+            - "@service_container" # lazy request resolution
```
Comment (docteurklein) [inline config/orm-services.yml]: - "%locale%"

why lazy? the parameter bag is frozen at build time, isn't it ?

Comment (jerome-fix) [inline config/orm-services.yml]: Something like that ? 

```yaml
    knp.doctrine_behaviors.translatable_subscriber.default_locale_callable:
        class:           "%knp.doctrine_behaviors.translatable_subscriber.default_locale_callable.class%"
        arguments:
            - %locale%
        public:  false
````

````php
class DefaultLocaleCallable
{
    private $locale;

    public function __construct($locale)
    {
        $this->locale = $locale;
    }

    public function __invoke()
    {
        return $this->locale;
    }
}
```

**Comment by docteurklein** [comment]
exactly!

**Comment by docteurklein** [comment]
(waiting for CI) and I'll merge

**Comment by jerome-fix** [comment]
@docteurklein could you merge this pull request or explain me what i should modify or adapt ?

**Comment by docteurklein** [comment]
thanks :)

**Comment by jerome-fix** [comment]
You welcome

Le 2015-03-23 09:25, Florian Klein a écrit :

**Comment by Rvanlaak** [comment]
This is nice, but can you add some documentation or a code sample about how to override the default locale? We get `nl_NL` as default locale, but we never make use of the full country locale anywhere. Did assume this was set to `en` as this is our framework fallback locale, but the `TranslatorSubscriber` thinks different about it.

**Comment by MAXakaWIZARD** [comment]
@jerome-fix @Rvanlaak  As I see, listeners are called, but finally locale gets determined as `en`, while it is set to `ru` in my config (in Translatable currentLocale is finally `null` and defaultLocale is `en`)

**Comment by MAXakaWIZARD** [comment]
Working solution is:
```php
/**
     * @ORM\PostUpdate
     * @ORM\PostPersist
     */
    public function updateSlug(LifecycleEventArgs $eventArgs)
    {
        $translatable = $this->getTranslatable();
        $em = $eventArgs->getEntityManager();
        $repo = $em->getRepository(get_class($translatable));
        //reload translatable to have current locale set
        $id = $translatable->getId();
        $em->detach($translatable);
        $translatable = $repo->find($id);

        $oldSlug = $translatable->getSlug();
        $translatable->generateSlug();

        if ($oldSlug !== $translatable->getSlug()) {
            $repo->updateSlug($translatable);
        }
    }
```
But I think it is quite cumbersome, maybe anyone can suggest better solution?

**Comment by MAXakaWIZARD** [comment]
Conclusion is: locale setting does not work for newly created objects. But this can be fixed by adding subscription to `prePersist` lifecycle callback. @docteurklein what do you think? I can make a PR for it.

**Pull request merged**